cid.go

```diff
@@ -168,15 +174,29 @@ func NewPrefixV1(codecType uint64, mhType uint64) Prefix {
 // Cid represents a self-describing content adressed
 // identifier. It is formed by a Version, a Codec (which indicates
 // a multicodec-packed content type) and a Multihash.
-type Cid struct {
-	version uint64
-	codec   uint64
-	hash    mh.Multihash
+// Byte layout: [version, codec, multihash]
+// - version uvarint
+// - codec uvarint
+// - hash mh.Multihash
+type Cid string
```

Comment:
Maybe we should make this `type Cid struct{ string }` to ensure the Cid type is always valid.